app_test_suite/steps/scenarios/upgrade.py

```diff
@@ -110,63 +115,84 @@ def pre_run(self, config: argparse.Namespace) -> None:
             self._upload_chart_to_app_catalog(config, stable_chart_file_path)
             app_catalog_cr = AppCatalogCR.objects(self._kube_client).get_by_name(TEST_APP_CATALOG_NAME)
             stable_ver_match = self._semver_regex_match.fullmatch(stable_chart_file_path)
             stable_app_version = cast(Match, stable_ver_match).group(1)
-            return stable_app_version, TEST_APP_CATALOG_NAME, app_catalog_cr.obj["spec"]["storage"]["URL"]
+            TestInfoProvider().extract_chart_info(stable_chart_file_path, CONTEXT_KEY_STABLE_CHART_YAML, context)
+            catalog_url = app_catalog_cr.obj["spec"]["storage"]["URL"]
+            chart_url = f"{catalog_url}/{app_name}-{stable_app_version}.tgz"
+            return stable_app_version, TEST_APP_CATALOG_NAME, catalog_url, chart_url
 
         catalog_url = get_config_value_by_cmd_line_option(config, key_cfg_stable_app_url)
         logger.info(f"Adding new app catalog named '{STABLE_APP_CATALOG_NAME}' with URL '{catalog_url}'.")
         app_catalog_cr = get_app_catalog_obj(STABLE_APP_CATALOG_NAME, catalog_url, self._kube_client)
         logger.debug(f"Creating AppCatalog '{app_catalog_cr.name}' with the stable app version.")
         app_catalog_cr.create()
 
-        stable_app_ver = get_config_value_by_cmd_line_option(config, key_cfg_stable_app_version)
-        if stable_app_ver == "latest":
-            stable_app_ver = self._get_latest_app_version(catalog_url, app_name)
-
-        return stable_app_ver, STABLE_APP_CATALOG_NAME, catalog_url
+        stable_chart_ver = get_config_value_by_cmd_line_option(config, key_cfg_stable_app_version)
+        if stable_chart_ver == "latest":
+            stable_chart_ver = self._get_latest_app_version(catalog_url, app_name)
+
+        chart_url = f"{catalog_url}/{app_name}-{stable_chart_ver}.tgz"
+        with TemporaryDirectory("-ats-download") as d:
+            try:
+                r = requests.get(chart_url, allow_redirects=True)
+                if not r.ok:
+                    raise ATSTestError(
+                        f"Error 'HTTP-{r.status_code}' when fetching remote chart '{chart_url}': {r.reason}"
+                    )
+            except RequestException as e:
+                logger.error(f"Error when trying to fetch remote chart '{chart_url}': '{e}'.")
+                raise
+            chart_file_name = os.path.join(d, "chart.tgz")
+            with open(chart_file_name, "wb") as f:
+                f.write(r.content)
+            TestInfoProvider().extract_chart_info(chart_file_name, CONTEXT_KEY_STABLE_CHART_YAML, context)
+        return stable_chart_ver, STABLE_APP_CATALOG_NAME, catalog_url, chart_url
 
     def run_tests(self, config: argparse.Namespace, context: Context) -> None:
         app_name = context[CONTEXT_KEY_CHART_YAML]["name"]
-        app_version = context[CONTEXT_KEY_CHART_YAML]["version"]
+        chart_version = context[CONTEXT_KEY_CHART_YAML]["version"]
```

Comment:
I'm confused by the naming here. Is this the version of the app CR or the upstream chart?

If its the version in the app CR I'd keep the current name.

Reply:
Ack. I was confused why this changed but it's the chart that is under test. So aligning with that makes sense.

Same with my other comments.